attributes/default.rb

```diff
@@ -2,57 +2,41 @@
 # attributes/mesosphere_packages.rb.
 default['mesos']['version']                                 = '0.20.1'
 
-# Configure Mesos master processes via environment variables here.
-# NOTE: Any attributes defined under ['mesos']['master']['env'] will be written
-# to /etc/default/mesos-master.
+#
+# Mesos MASTER configuration
+#
 
 # Ulimit to pass to mesos-master process.
 default['mesos']['master']['env']['ULIMIT']                 = '-n 16384'
 
-# Directory to write mesos-master logs.
-default['mesos']['master']['env']['LOGS']                   = '/var/log/mesos'
-
-# Configure Mesos via command line options here.
-# NOTE: Any attributes defined under ['mesos']['master']['flags'] will be
-# written as individual files within the directory /etc/mesos-master/.
-# Mesos init scripts will then use file in that directory as key/value pairs
-# to pass along as command line options to the mesos-master process.
-
 # mesos-master default port
 default['mesos']['master']['flags']['port']                 = 5050
 
+# Directory to write mesos-master logs.
+default['mesos']['master']['flags']['log_dir']              = '/var/log/mesos'
+
 # mesos-master default logging level.  Valid options: 'INFO', 'WARNING', 'ERROR']
 default['mesos']['master']['flags']['logging_level']        = 'INFO'
 
 # mesos-master cluster name that is visible from UI.
 default['mesos']['master']['flags']['cluster']              = 'MyMesosCluster'
 
-# ZooKeeper URL (used for leader election amongst masters).
-# In this case we are deferring zk URL to what is in /etc/mesos/zk which this cookbook
-# manages based on zookeeper specific attributes below.
-default['mesos']['master']['flags']['zk']                   = '`cat /etc/mesos/zk`'
```

Comment:
If we remove this flag then the master has no way of locating the zookeeper cluster that it needs to register with.